### PR TITLE
fix: avoid getting crashed LiveSync when the fast sync fails

### PR DESCRIPTION
**Pull request by DimitarTachev**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/nativescript-cli/blob/master/CONTRIBUTING.md#contribute-to-the-code-base
- [ ] Tests for the changes are included.

In short, the LiveSync crashes when the App is not started or installed on the device.

* Re-implements: https://github.com/NativeScript/nativescript-cli/pull/4020
* Imporves `tns test android` LiveSync performance (avoid waiting 30 seconds)

Related to: https://github.com/NativeScript/nativescript-cli/issues/4914
https://github.com/NativeScript/nativescript-cli/issues/4916

### CODE REVIEW

**Comment by KristianDD** [comment]
test cli-smoke cli-run cli-templates package_version#rc

**Comment by DimitarTachev** [comment]
test cli-run package_version#rc
